components/Dialogs/src/MDCDialogTransitionController.m

```diff
@@ -60,30 +71,60 @@ - (void)animateTransition:(id<UIViewControllerContextTransitioning>)transitionCo
       UIViewAnimationOptionAllowUserInteraction | UIViewAnimationOptionBeginFromCurrentState;
 
-  [UIView animateWithDuration:transitionDuration
-      delay:0.0
-      options:options
-      animations:^{
-        animatingView.alpha = endingAlpha;
-      }
-      completion:^(__unused BOOL finished) {
-        // If we're dismissing, remove the presented view from the hierarchy
-        if (!presenting) {
-          [fromView removeFromSuperview];
-        }
-
-        // From ADC : UIViewControllerContextTransitioning
-        // When you do create transition animations, always call the
-        // completeTransition: from an appropriate completion block to let UIKit know
-        // when all of your animations have finished.
-        [transitionContext completeTransition:YES];
-      }];
+  // Opacity animation
+  animatingView.alpha = startingAlpha;
+  [UIView animateWithDuration:_animationDuration / 2
+                        delay:0
+                      options:options
+                   animations:^{
+                     animatingView.alpha = endingAlpha;
+                   }
+                   completion:nil];
+
+  // Scale animation
+  MDCDialogPresentationController *presentationController = nil;
+  if ([animatingViewController.presentationController
+          isKindOfClass:[MDCDialogPresentationController class]]) {
+    presentationController =
+        (MDCDialogPresentationController *)animatingViewController.presentationController;
+  }
+
+  CGAffineTransform startingTransform =
+      presenting ? CGAffineTransformMakeScale(_dialogScaleFactor, _dialogScaleFactor)
+                 : CGAffineTransformIdentity;
+  CGAffineTransform endingTransform = CGAffineTransformIdentity;
+  animatingView.transform = startingTransform;
+  presentationController.dialogTransform = startingTransform;
+  UIViewAnimationOptions scaleAnimationOptions = options | UIViewAnimationOptionCurveEaseOut;
+  [UIView animateWithDuration:_animationDuration
+                        delay:0
+                      options:scaleAnimationOptions
+                   animations:^{
+                     animatingView.transform = endingTransform;
+                     presentationController.dialogTransform = endingTransform;
+                   }
+                   completion:nil];
+
+  // Use dispatch_after to avoid animateWithDuration immediately executing its completion block.
+  // This can happen if the animation has no effect (changing animatingView.transform from
+  // CGAffineTransformIdentity to CGAffineTransformIdentity, in this case).
+  dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(_animationDuration * NSEC_PER_SEC)),
+                 dispatch_get_main_queue(), ^{
+                   // If we're dismissing, remove the presented view from the hierarchy
+                   if (!presenting) {
+                     [fromView removeFromSuperview];
+                   }
+
+                   // From ADC : UIViewControllerContextTransitioning
```

Comment:
Tip: I don't know what "ADC" is, but providing a link to Apple's docs would be helpful:

https://developer.apple.com/documentation/uikit/uiviewcontrollercontexttransitioning

Reply:
I'm not sure either, to be honest. This comment was here before I made the change and it's showing up as changed because I moved the completion block.